mpas_analysis/ocean/histogram.py

```diff
@@ -0,0 +1,589 @@
+                    plotRegionSubtask.run_after(obsMasksSubtask)
+                    self.add_subtask(plotRegionSubtask)
+
+    def setup_and_check(self):
+        """
+        Perform steps to set up the analysis and check for errors in the setup.
+
+        Raises
+        ------
+        OSError
+            If files are not present
+        """
+        # first, call setup_and_check from the base class (AnalysisTask),
+        # which will perform some common setup, including storing:
+        #   self.inDirectory, self.plotsDirectory, self.namelist, self.streams
+        #   self.calendar
+        super().setup_and_check()
+
+        # Add variables and seasons to climatology task
+        variableList = []
+        for var in self.variableList:
+            variableList.append(f'timeMonthly_avg_{var}')
+
+        self.mpasClimatologyTask.add_variables(variableList=variableList,
+                                               seasons=self.seasons)
+
+        if self.weightList is not None:
+            if len(self.weightList) != len(self.variableList):
+                raise ValueError('Histogram weightList is not the same '
+                                 'length as variableList')
+        if len(self.obsList) > 1:
+            raise ValueError('Histogram analysis does not currently support'
+                             'more than one observational product')
+
+
+class ComputeHistogramWeightsSubtask(AnalysisTask):
+    """
+    Fetches weight variables from MPAS output files for each variable in
+    variableList.
+
+    """
+    def __init__(self, parentTask, regionName, mpasMasksSubtask, fullSuffix,
+                 variableList, weightList):
+        """
+        Initialize weights task
+
+        Parameters
+        ----------
+        parentTask :  ``AnalysisTask``
+            The parent task, used to get the ``taskName``, ``config`` and
+            ``componentName``
+
+        regionName : str
+            Name of the region to plot
+
+        mpasMasksSubtask : ``ComputeRegionMasksSubtask``
+            A task for creating mask MPAS files for each region to plot, used
+            to get the mask file name
+
+        obsDicts : dict of dicts
+            Information on the observations to compare agains
+
+        fullSuffix : str
+            The regionGroup and regionName combined and modified to be
+            appropriate as a task or file suffix
+
+        variableList: list of str
+            List of variables which will be weighted
+
+        weightList: list of str
+            List of variables by which to weight the variables in
+            variableList, of the same length as variableList
+
+        """
+
+        super(ComputeHistogramWeightsSubtask, self).__init__(
+            config=parentTask.config,
+            taskName=parentTask.taskName,
+            componentName=parentTask.componentName,
+            tags=parentTask.tags,
+            subtaskName=f'weights{fullSuffix}_{regionName}')
```

Comment:
Since the outer loop is over region group and then region name, I think this needs to be something like:
```suggestion
            subtaskName=f'weights_{regionGroupSuffix}_{regionName}')
```
`regionGroupSuffix` will need to be passed in as an argument.

The `fullSuffix` would also work with my suggested change above, but has the redundant `histogram_` prefix.